lib/xandra/cluster/pool.ex

```diff
@@ -60,7 +60,8 @@ defmodule Xandra.Cluster.Pool do
     :gen_statem.stop(pid, reason, timeout)
   end
 
-  @spec checkout(:gen_statem.server_ref()) :: {:ok, pid()} | {:error, :empty}
+  @spec checkout(:gen_statem.server_ref()) ::
+          {:ok, nonempty_list({pid(), Host.t()})} | {:error, :empty}
```

Comment:
Nitpick: I prefer this syntax for compactness:

```suggestion
          {:ok, [{pid(), Host.t()}, ...]} | {:error, :empty}
```